EasyPost.Tests/ShipmentTest.cs

```diff
@@ -230,5 +267,56 @@ public async Task TestCreateWithIds()
             Assert.IsTrue(shipment.parcel.id.StartsWith("prcl_"));
             Assert.AreEqual("388 Townsend St", shipment.from_address.street1);
         }
+
+        [TestMethod]
+        public async Task TestLowestRate()
+        {
+            _vcr.SetUpTest("lowest_rate");
+
+            Shipment shipment = await CreateFullShipment();
+
+            // test lowest rate with no filters
+            Rate lowestRate = shipment.LowestRate();
+            Assert.AreEqual("First", lowestRate.service);
+            Assert.AreEqual("5.49", lowestRate.rate);
+            Assert.AreEqual("USPS", lowestRate.carrier);
+
+            // test lowest rate with service filter (this rate is higher than the lowest but should filter)
+            List<string> services = new List<string>
+            {
+                "Priority"
+            };
+            lowestRate = shipment.LowestRate(null, services, null, null);
+            Assert.AreEqual("Priority", lowestRate.service);
+            Assert.AreEqual("7.37", lowestRate.rate);
+            Assert.AreEqual("USPS", lowestRate.carrier);
+
+            // test lowest rate with carrier filter (should error due to bad carrier)
+            List<string> carriers = new List<string>
+            {
+                "BAD_CARRIER"
+            };
+            Assert.ThrowsException<FilterFailure>(() => shipment.LowestRate(carriers, null, null, null));
+        }
+
+        [TestMethod]
+        public async Task TestLowestSmartrate()
+        {
+            _vcr.SetUpTest("lowest_smartrate");
+
+            Shipment shipment = await CreateBasicShipment();
+
+            // test lowest smartrate with valid filters
+            Smartrate lowestSmartrate = await shipment.LowestSmartrate(1, SmartrateAccuracy.Percentile90);
+            Assert.AreEqual("Priority", lowestSmartrate.service);
+            Assert.AreEqual(7.37, lowestSmartrate.rate);
+            Assert.AreEqual("USPS", lowestSmartrate.carrier);
+
+            // test lowest smartrate with invalid filters (should error due to strict delivery_days)
+            await Assert.ThrowsExceptionAsync<FilterFailure>(async () => await shipment.LowestSmartrate(0, SmartrateAccuracy.Percentile90));
+
+            // test lowest smartrate with invalid filters (should error due to bad delivery_accuracy)
+            // this test is not needed in the C# CL because it uses enums for the accuracy (can't pass in an incorrect value)
```

Comment:
Right, we want to test that your enums work is the point. This will guard against enums getting removed/changed in the future by someone without context.